nucypher/blockchain/eth/interfaces.py

```diff
@@ -108,6 +110,45 @@ class UnknownContract(InterfaceError):
     class NotEnoughConfirmations(InterfaceError):
         pass
 
+    REASONS = {
+        INSUFFICIENT_ETH: 'insufficient funds for gas * price + value',
+    }
+
+    class TransactionFailed(InterfaceError):
+
+        IPC_CODE = -32000  # (geth)
+
+        def __init__(self,
+                     message: str,
+                     transaction_dict: dict,
+                     contract_function: Union[ContractFunction, ContractConstructor],
+                     *args):
+
+            self.base_message = message
+            self.name = get_transaction_name(contract_function=contract_function)
+            self.payload = transaction_dict
+            self.contract_function = contract_function
+            self.failures = {
```

Comment:
Would this be better served as a class variable instead of being created per object?

Reply:
^ @KPrasch one outstanding suggestion from me.

Reply:
Yeah - it's totally natural and I agree - but I cannot get the other signer classes in scope without breaking this module into a package. and juggling circular imports :-(  

Perhaps give it a shot? :-)